File: src/main/java/com/thesilentnights/easylogin/commands/EasyLoginCommands.java
package com.thesilentnights.easylogin.commands;

import com.mojang.brigadier.CommandDispatcher;
import com.thesilentnights.easylogin.commands.common.Login;
import net.minecraft.commands.CommandSourceStack;


public class EasyLoginCommands {
    private final Login login;


    public void register(CommandDispatcher<CommandSourceStack> dispatcher) {

        // Common commands
        dispatcher.register();
        dispatcher.register(koin.<Registrar>get().getCommand());
        dispatcher.register(koin.<Logout>get().getCommand());
        dispatcher.register(koin.<ChangePassword>get().getCommand());
        dispatcher.register(koin.<Email>get().getCommand());
        dispatcher.register(koin.<Recover>get().getCommand());

        // Admin commands
        dispatcher.register(koin.<ByPass>get().getCommand(AdminCommands.mainNode));
        dispatcher.register(koin.<TeleportToOfflinePlayer>get().getCommand(AdminCommands.mainNode));
        dispatcher.register(koin.<EmailTest>get().getCommand(AdminCommands.mainNode));
        dispatcher.register(koin.<PlayerInfoCommands>get().getCommand(AdminCommands.mainNode));
    }
}

File: src/main/kotlin/com/thesilentnights/easylogin/commands/EasyLoginCommands.kt
package com.thesilentnights.easylogin.commands

import com.mojang.brigadier.CommandDispatcher
import com.thesilentnights.easylogin.commands.server.admin.*
import com.thesilentnights.easylogin.commands.server.common.*
import net.minecraft.commands.CommandSourceStack

object EasyLoginCommands {
    fun register(dispatcher: CommandDispatcher<CommandSourceStack>) {
        //common
        with(dispatcher) {
            register(Login().command)
            register(Registrar().command)
            register(ChangePassword().command)
            register(Email().command)
            register(Logout().command)
            register(Recover().command)
        }

        //admin
        with(dispatcher) {
            with(AdminCommands) {
                register(ByPass().getCommand(mainNode))
                register(PlayerInfoCommands().getCommand(mainNode))
                register(TeleportToOfflinePlayer().getCommand(mainNode))
                register(EmailTest().getCommand(mainNode))
            }
        }
    }

}
